hazelcast/src/main/java/com/hazelcast/config/EvictionConfig.java

```diff
@@ -287,7 +282,7 @@ public final boolean equals(Object o) {
 
         EvictionConfig that = (EvictionConfig) o;
 
-        return size == that.size
+        return ((size == -1 && that.size == -1) || getSize() == that.getSize())
```

Comment:
Writing like this is seemed more open: 


```suggestion
        ((size == -1 && that.size == -1) || defaultSize == that.defaultSize)
```

Reply:
But this does not mean the same thing. 
This should hold true either
- when both of the fields are unset. It does not matter if they are accessed via `NearCacheConfigAccessor` or not, it should hold true. 
Or
- when at least a field is set, and the effective sizes are equal. 


Your suggestion is confusing to me. For example, the following simple test fails when I apply it.
```
	    EvictionConfig evictionConfig = new EvictionConfig().setSize(4);
        EvictionConfig evictionConfig2 = new EvictionConfig().setSize(5);

        assertNotEquals(evictionConfig, evictionConfig2);
```

Reply:
I see you are right. 
What about `size == that.size && defaultSize == that.defaultSize` instead?

Reply:
In that case, the new test does not pass. 
```
    public void testNearCacheConfigEqual_BeforeAndAfterDefaultSizeIsInitialized() {
        NearCacheConfig config1 = new NearCacheConfig();
        NearCacheConfig config2 = new NearCacheConfig();
        NearCacheConfigAccessor.initDefaultMaxSizeForOnHeapMaps(config1);

        assertEquals(config1, config2);
    }
```